lib/min_heap.rb

```diff
@@ -14,21 +14,33 @@ def initialize
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: 0(1)?
+  # Space Complexity: o(n)?
   def add(key, value = key)
```

Comment:
👍 The time complexity would be O(log n) because of how you heap_up going up through the levels of the tree.

Similarly because `heap_up` is recursive the time complexity is O(log n).